app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -785,7 +786,13 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         // has removed all of them, or we couldn't load any) we will pass searchTermOrURL to Gecko
         // and let it try to load whatever was entered.
         if ((!forceSearch && searchTermOrURL.isUrl()) || engine == null) {
-            loadUrlUseCase.invoke(searchTermOrURL.toNormalizedUrl(), flags)
+            if (loadUrlUseCase is SessionUseCases.DefaultLoadUrlUseCase) {
```

Comment:
Thanks for taking this on! Can you describe the fix and maybe add a comment here too? It's not clear to me reading the patch why this fixes the issue.

Since this is an important fix, it would great to cover with a test too in `HomeActivityTest`...could break into a separate method to make it easier to test.

Reply:
@mcarare 😱  So you're seeing the PWA session being "selected" in `SessionManager`, but (correctly) not selected in the `Store`? The custom tab and PWA sessions should never be marked as selected. Can you reproduce this reliably?

If so, we should get to the bottom of this and see how the wrong session gets selected, as this can cause a number of different problems. If we fail and can't find the cause, we can still land the workaround, but we should definitely try.

/cc @pocmo

Reply:
@csadilek Yes, I think the problem lies in `sessionManager` and the way the selected session is updated, not exclusively in relation to PWA or custom tabs. 
Maybe is related to another issue fixed in another of my  PRs: https://github.com/mozilla-mobile/fenix/pull/17412. If you open a tab from collection it is not automatically selected, but it should have been by simply opening a new tab with the respective restored tab session.

Reply:
And yes, I can reproduce this 100% of the time.

Reply:
Tested some more, this only happens with PWA it does not happen with regular not custom tabs.

Reply:
OK, it seems you have reliable STR? Next step is to debug `LegacySessionManager` and see how the PWA tab becomes the selected tab. I think that's the part we need to understand to fix this in all cases. How can that happen?

Reply:
`WebAppIntentProcessor` seems to be the problem. I will open an issue and a PR on AC.